src/org/rascalmpl/library/Prelude.java

```diff
@@ -3948,30 +3950,39 @@ private static final class ReleasableCallback implements Consumer<ISourceLocatio
 		private final int hash;
 
 		private final IValueFactory values;
-		private final Type deleted;
-		private final Type created;
-		private final Type modified;
-		private final Type file;
-		private final Type directory;
-		private final Type changeEvent;
 
+		private static final Type deleted;
+		private static final Type created;
+		private static final Type modified;
+		private static final Type file;
+		private static final Type directory;
+		private static final Type changeEvent;
+		static {
```

Comment:
I have some worries here. The static block will not be initialized in a thread-safe or thread-friendly way. So haven't we introduced more races than we have solved here?

Reply:
How can this introduce a race? [Static initializers are thread safe](https://stackoverflow.com/questions/878577/are-java-static-initializers-thread-safe)

Reply:
I've rewritten them, it was a premature optimization to keep them in static fields.